packages/react/src/use-chat.ts

```diff
@@ -183,45 +190,68 @@ Default is undefined, which disables throttling.
   const chatId = id ?? hookId;
   const chatKey = typeof api === 'string' ? [api, chatId] : chatId;
 
-  // Store array of the processed initial messages to avoid re-renders:
-  const stableInitialMessages = useStableValue(initialMessages ?? []);
-  const processedInitialMessages = useMemo(
-    () => stableInitialMessages,
-    [stableInitialMessages],
+  const chatStore = useRef(
+    store ??
+      new ChatStore({
+        chats: {
+          [chatId]: {
+            messages: [],
+          },
+        },
+        '~internal': {
+          currentDate: getCurrentDate,
+        },
+      }),
   );
 
-  // Store the chat state in SWR, using the chatId as the key to share states.
-  const { data: messages, mutate } = useSWR<UIMessage[]>(
-    [chatKey, 'messages'],
-    null,
-    { fallbackData: processedInitialMessages },
-  );
+  if (!chatStore.current.hasChat(chatId)) {
+    chatStore.current.addChat(chatId, []);
+  }
 
-  // Keep the latest messages in a ref.
-  const messagesRef = useRef<UIMessage[]>(messages || []);
-  useEffect(() => {
-    messagesRef.current = messages || [];
-  }, [messages]);
+  const { subscribe, getMessages, getStatus, getError, setStatus } =
+    useChatStore({
+      store: chatStore.current,
+    });
```

Comment:
why is this necessary vs. using chatStore current directly?

Reply:
i'm leaning towards this so that we can encapsulate the stable selector functions (e.g. `getMessages`) in one spot! like a separation of concerns b/w subscription/reading (within hook) and imperative mutations